testsuite/shared/src/main/java/org/jboss/as/test/deployment/DeploymentInfoUtils.java

```diff
@@ -0,0 +1,576 @@
+/*
+Copyright 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.deployment;
+
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.cli.CommandFormatException;
+import org.jboss.as.test.integration.management.base.AbstractCliTestBase;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.wildfly.common.annotation.NotNull;
+
+import java.io.IOException;
+
+import static org.hamcrest.core.AllOf.allOf;
+import static org.jboss.as.cli.Util.FAILED;
+import static org.jboss.as.cli.Util.FAILURE_DESCRIPTION;
+import static org.jboss.as.cli.Util.OUTCOME;
+import static org.jboss.as.cli.Util.RESULT;
+import static org.jboss.as.cli.Util.SUCCESS;
+import static org.jboss.as.test.deployment.DeploymentInfoUtils.DeploymentState.NOT_ADDED;
+import static org.jboss.as.test.deployment.DeploymentInfoUtils.DeploymentState.UNKNOWN;
+import static org.jboss.as.test.deployment.DeploymentInfoUtils.DeploymentState.mapBooleanByDeploymentStatus;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.fail;
+
+/**
+ * Utils for verify state of applications deployments using command 'deployments list' and 'deployments info'.
+ * Uses legacy and Aesh version of commands.
+ * Direct verifying output of this commands.
+ * Support double check verification in case set CommandContext by management command.
+ *
+ * @author Vratislav Marek (vmarek@redhat.com)
+ **/
+public class DeploymentInfoUtils extends AbstractCliTestBase {
+    private static final String OUTPUT_EMPTY_MARK = "--OUTPUT EMPTY--";
+
+    private static final Logger log = Logger.getLogger(DeploymentInfoUtils.class);
+
+    // Ip address for connection to cli by 'AbstractCliTestBase.initCLI(ipAddress)'
+    private String ipAddress;
+    // In case you disabled double checking, this variable is for enable double checking without parameter
+    private CommandContext disabledCtx;
+    private CommandContext ctx;
+
+    // Holding output of last time called command for multiple checking output without recall command
+    private String[] currentOutputRows;
```

Comment:
It would be better to have content returned by methods and not referenced from the instance. In general, it could be better to call a command two times (it is not that costly) instead to keep a state. Having to deal with state could introduce bugs in the tests.